Add coretime chainspecs for kusama and polkadot and presets for live chains

seadanda opened the pull request:
Add the invulnerables set directly to the preset and generate chain-specs for the "live" (production) networks directly from the `chain-spec-builder` instead of using a bash script to post-process these from generated local chain-specs.

Kusama is added here for completeness and a sanity check with the original genesis chain-specs.

Note that the SS58 prefix is ignored when generating the chainspecs, as noted in https://github.com/paritytech/polkadot-sdk/issues/5271

This can be used as a merge target for community boot nodes for the Polkadot Coretime chain to be added before genesis by adding your server to the array in the `coretime_polkadot_config`: https://github.com/seadanda/runtimes/blob/6ba4bc747b9fc547b0ca9192f6b89b43c2e70ed2/chain-spec-generator/src/system_parachains_specs.rs#L292

seadanda committed:
Add kusama live preset and chain spec to builder

seadanda committed:
Add coretime-polkadot chainspec

seadanda committed:
Merge branch 'polkadot-fellows:main' into donal-coretime-chain-specs

seadanda committed:
Make clippy happy

seadanda committed:
Remove unnecessary dep

michalkucharczyk commented on chain-spec-generator/src/system_parachains_specs.rs:
```diff
@@ -218,6 +220,45 @@ pub fn coretime_kusama_local_testnet_config() -> Result<Box<dyn ChainSpec>, Stri
+		.with_name("Kusama Coretime")
+		.with_id("coretime-kusama")
+		.with_chain_type(ChainType::Live)
+		.with_genesis_config_patch(
```
You could use: [`with_genesis_config_preset_name`](https://github.com/paritytech/polkadot-sdk/blob/6db203822418eee666f44ccaed3cd96fdfa27351/substrate/client/chain-spec/src/chain_spec.rs#L404).

This would allow to fetch the preset from the wasm blob, w/o necessity to natively compile the runtime,

seadanda replied:
Nice, didn't know that existed. Thanks!

michalkucharczyk commented on chain-spec-generator/src/system_parachains_specs.rs:
```diff
@@ -242,6 +283,45 @@ pub fn coretime_polkadot_local_testnet_config() -> Result<Box<dyn ChainSpec>, St
+		.with_name("Polkadot Coretime")
+		.with_id("coretime-polkadot")
+		.with_chain_type(ChainType::Live)
+		.with_genesis_config_patch(
```
also here  [`with_genesis_config_preset_name`](https://github.com/paritytech/polkadot-sdk/blob/6db203822418eee666f44ccaed3cd96fdfa27351/substrate/client/chain-spec/src/chain_spec.rs#L404) should do the job w/o native runtime dep.

seadanda committed:
Use presets for new chainspecs

seadanda committed:
Merge branch 'donal-coretime-polkadot' into donal-coretime-chain-specs

seadanda committed:
Merge branch 'donal-coretime-polkadot' into donal-coretime-chain-specs

seadanda committed:
Add live to preset names

senseless committed:
Add stake plus bootnodes to coretime polkadot at release. (#5)

seadanda committed:
Merge branch 'donal-coretime-polkadot' into donal-coretime-chain-specs

seadanda committed:
Add the Polkadot Coretime runtime (#410)

Add the Polkadot Coretime chain in advance of the 1.3.0 release.

This uses the new Price adapter which has been running on Kusama now for
two sales cycles, and includes the mechanism to burn revenue.

TODO:
- [x] Add Transact tests for hardcoded weights after
https://github.com/polkadot-fellows/runtimes/pull/401 is merged
- [ ] Rerun benchmarks and check hardcoded weights after merge

The genesis chain-spec is developing on
https://github.com/seadanda/runtimes/pull/4. This can be used as a merge
target for any community boot nodes who would like to be included at
genesis and will be separately merged to `main`.

---------

Co-authored-by: Bastian Köcher <git@kchr.de>
Co-authored-by: Branislav Kontur <bkontur@gmail.com>
Co-authored-by: fellowship-merge-bot[bot] <151052383+fellowship-merge-bot[bot]@users.noreply.github.com>

seadanda committed:
Merge branch 'main' into donal-coretime-chain-specs

seadanda committed:
Update changelog

acatangiu committed:
all runtimes: remove already applied migrations (#420)

Remove migrations already **applied on-chain**.

- [x] Does not require a CHANGELOG entry

---------

Signed-off-by: Adrian Catangiu <adrian@parity.io>
Co-authored-by: joe petrowski <25483142+joepetrowski@users.noreply.github.com>

seadanda committed:
Merge branch 'main' into donal-coretime-chain-specs

fellowship-merge-bot[bot] committed:
Merge branch 'main' into donal-coretime-chain-specs
